server/sonar-server/src/main/java/org/sonar/server/component/ws/ProjectMeasuresQueryValidator.java

```diff
@@ -19,73 +19,48 @@
  */
 package org.sonar.server.component.ws;
 
-import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashSet;
-import java.util.List;
 import java.util.Set;
-import java.util.TreeSet;
-import org.sonar.db.DbClient;
-import org.sonar.db.DbSession;
-import org.sonar.db.metric.MetricDto;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
 import org.sonar.server.measure.index.ProjectMeasuresQuery;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Arrays.asList;
 import static org.sonar.core.util.stream.MoreCollectors.toHashSet;
-import static org.sonar.core.util.stream.MoreCollectors.toSet;
+import static org.sonar.db.measure.ProjectMeasuresIndexerIterator.METRIC_KEYS;
 import static org.sonar.server.measure.index.ProjectMeasuresQuery.MetricCriterion;
 import static org.sonar.server.measure.index.ProjectMeasuresQuery.SORT_BY_LAST_ANALYSIS_DATE;
 import static org.sonar.server.measure.index.ProjectMeasuresQuery.SORT_BY_NAME;
 
 public class ProjectMeasuresQueryValidator {
 
-  private static final Set<String> NON_METRIC_SORT_KEYS = new HashSet<>(asList(SORT_BY_NAME, SORT_BY_LAST_ANALYSIS_DATE));
+  static final Set<String> NON_METRIC_SORT_KEYS = new HashSet<>(asList(SORT_BY_NAME, SORT_BY_LAST_ANALYSIS_DATE));
 
-  private final DbClient dbClient;
-
-  public ProjectMeasuresQueryValidator(DbClient dbClient) {
-    this.dbClient = dbClient;
+  private ProjectMeasuresQueryValidator() {
   }
 
-  public void validate(DbSession dbSession, ProjectMeasuresQuery query) {
-    Set<String> metricKeys = getMetrics(query);
-    if (metricKeys.isEmpty()) {
-      return;
-    }
-    List<MetricDto> dbMetrics = dbClient.metricDao().selectByKeys(dbSession, new ArrayList<>(metricKeys));
-    checkMetricKeysExists(dbMetrics, metricKeys);
-    checkMetricsAreEnabled(dbMetrics);
-    checkMetricsAreNumerics(dbMetrics);
+  public static void validate(ProjectMeasuresQuery query) {
```

Comment:
Please remove ProjectMeasuresQueryValidator from pico as it's not used as static method.